internal/logger/default_logger.go

```diff
@@ -21,124 +21,122 @@ import (
+		return false
 	}
+	return loggerLevel >= numericLevel
 }
 
-func (l *DefaultLogger) Infof(format string, values ...interface{}) {
-	if l.canLogInfo() {
-		s := fmt.Sprintf("INFO : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) Log(level logger.Level, formatter func() string) {
+	if !l.canLog(level) || formatter == nil {
+		return
 	}
+	s := fmt.Sprintf("%s: %s", strings.ToUpper(level.String()), formatter())
+	_ = l.Output(logCallDepth, s)
 }
 
-// Warn logs the given arguments at warn level if the level is greater than or equal to warn level.
-func (l *DefaultLogger) Warn(args ...interface{}) {
-	if l.canLogWarn() {
-		callerName := l.findCallerFuncName()
-		s := callerName + "\n" + warnPrefix + ": " + fmt.Sprint(args...)
-		l.Output(logCallDepth, s)
-	}
+func (l *DefaultLogger) findCallerFuncName() string {
+	pc, _, _, _ := runtime.Caller(logCallDepth)
+	return runtime.FuncForPC(pc).Name()
 }
 
-func (l *DefaultLogger) Warnf(format string, values ...interface{}) {
-	if l.canLogWarn() {
-		s := fmt.Sprintf("WARN : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
-	}
+// LogAdaptor is used to convert logger implementations of public interface logger.LogAdaptor to internal logging interface LogAdaptor
+type LogAdaptor struct {
+	logger.Logger
 }
 
-// Error logs the given arguments at error level if the level is greater than or equal to error level.
-func (l *DefaultLogger) Error(err error) {
-	l.Errorf(err.Error())
+// Debug runs the given function to generate the logger string, if logger level is debug or finer.
+func (la LogAdaptor) Debug(f func() string) {
+	la.Log(logger.DebugLevel, f)
 }
 
-func (l *DefaultLogger) Errorf(format string, values ...interface{}) {
-	if l.canLogError() {
-		s := fmt.Sprintf("ERROR: %s", fmt.Errorf(format, values...).Error())
-		l.Output(logCallDepth, s)
-	}
+// Trace runs the given function to generate the logger string, if logger level is trace or finer.
+func (la LogAdaptor) Trace(f func() string) {
+	la.Log(logger.TraceLevel, f)
 }
 
-func (l *DefaultLogger) findCallerFuncName() string {
-	pc, _, _, _ := runtime.Caller(logCallDepth)
-	return runtime.FuncForPC(pc).Name()
+// Info runs the given function to generate the logger string, if logger level is info or finer.
+func (la LogAdaptor) Info(f func() string) {
+	la.Log(logger.InfoLevel, f)
 }
 
-func (l *DefaultLogger) canLogTrace() bool {
-	return l.Level >= traceLevel
+// Infof formats the given string with the given values, if logger level is info or finer.
+func (la LogAdaptor) Infof(format string, values ...interface{}) {
+	la.Log(logger.InfoLevel, func() string {
+		return fmt.Sprintf(format, values...)
+	})
 }
 
-func (l *DefaultLogger) canLogInfo() bool {
-	return l.Level >= infoLevel
+// Warn formats the given string with the given values, if logger level is warn or finer.
+func (la LogAdaptor) Warn(f func() string) {
+	la.Log(logger.WarnLevel, f)
```

Comment:
This `Warn` function does something different than the old one. If it's not being used, let's remove it.